src/lib.rs

```diff
@@ -0,0 +1,1210 @@
+    }
+}
+
+impl<'a, T: Debug + One + PrimInt + Zero> ExactSizeIterator for Iter<'a, T> { }
+
+impl<'a, T: Debug + One + PrimInt + Zero> IntoIterator for &'a Vob<T> {
+    type Item = bool;
+    type IntoIter = Iter<'a, T>;
+
+    fn into_iter(self) -> Iter<'a, T> {
+        self.iter()
+    }
+}
+
+#[derive(Clone)]
+pub struct IterSetBits<'a, T: 'a> {
+    vob: &'a Vob<T>,
+    range: Range<usize>,
+}
+
+impl<'a, T: Debug + One + PrimInt + Zero> Iterator for IterSetBits<'a, T> {
+    type Item = usize;
+
+    fn next(&mut self) -> Option<usize> {
+        if let Some(mut i) = self.range.next() {
+            for b in block_offset::<T>(i) .. blocks_required::<T>(self.vob.len) {
+                let v = self.vob.vec[b];
+                if v != T::zero() {
+                    let i_off = i % bits_per_block::<T>();
+                    let tz = (v >> i_off).trailing_zeros() as usize;
+                    if tz < bits_per_block::<T>() {
+                        let bs = b * bits_per_block::<T>() + i_off + tz;
+                        self.range.start = bs + 1;
+                        return Some(bs);
+                    }
+                }
+                i = b * bits_per_block::<T>();
+            }
+        }
+        self.range.start = self.range.end;
+        None
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.range.size_hint()
+    }
+}
+
+#[derive(Clone)]
+pub struct IterUnsetBits<'a, T: 'a> {
+    vob: &'a Vob<T>,
+    range: Range<usize>,
+}
+
+impl<'a, T: Debug + One + PrimInt + Zero> Iterator for IterUnsetBits<'a, T> {
+    type Item = usize;
+
+    fn next(&mut self) -> Option<usize> {
+        if let Some(mut i) = self.range.next() {
+            for b in block_offset::<T>(i) .. blocks_required::<T>(self.vob.len) {
+                let v = self.vob.vec[b];
+                if v != T::max_value() {
+                    let i_off = i % bits_per_block::<T>();
+                    let tz = (!v >> i_off).trailing_zeros() as usize;
+                    if tz < bits_per_block::<T>() {
+                        let bs = b * bits_per_block::<T>() + i_off + tz;
+                        self.range.start = bs + 1;
+                        if bs >= self.vob.len {
+                            break;
+                        }
+                        return Some(bs);
+                    }
+                }
+                i = b * bits_per_block::<T>();
+            }
+        }
+        self.range.start = self.range.end;
+        None
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.range.size_hint()
+    }
+}
+
+impl<T: Debug + One + PrimInt + Zero> PartialEq for Vob<T> {
+    fn eq(&self, other: &Self) -> bool {
+        if self.len != other.len {
+            return false;
+        }
+        self.iter_storage()
+            .zip(other.iter_storage())
+            .all(|(w1, w2)| w1 == w2)
+    }
+}
+
+impl<T: Debug + One + PrimInt + Zero> Eq for Vob<T> {}
```

Comment:
I have to admit, I don't fully understand `Eq` and `PartialEq` in rust.

Looks to me like we have a total ordering here, so I'd expect to implement `Eq` not `PartialEq`, but `Eq` has no methods.

The docs didn't manage to explain it to me so well. I'll look into it more some time.

Reply:
You have to implement `PartialEq` in order to be able to implement `Eq` (but the latter doesn't require additional methods). It is a bit weird, but it does eventually make sense.

Reply:
OK, you'll have to explain that one to me one day :)

Reply:
Basically they're special-cased in the compiler. Mostly it's not a very important detail IME.